src/components/PrivateRoute/PrivateRoute.jsx

```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import CourseForm from '../CourseForm/CourseForm';
+
+const PrivateRoute = ({ children }) => {
+  const user = useSelector((state) => state.courses.user);
+
+  if (user.role === 'admin') {
+    return <CourseForm />;
+  } else return children;
```

Comment:
Well, not really agree with such condition: whenever you use private routes, you should redirect user to default/login page, not just rendering this component. It might cause confusion for user.
Let's use `Navigate` component and perform redirection to corresponding route